libs/sdk-core/src/swap_in/swap.rs

```diff
@@ -393,49 +394,57 @@ impl BTCReceiveSwap {
-                .payment_receiver
-                .receive_payment(ReceivePaymentRequest {
-                    amount_msat: swap_info.confirmed_sats * 1000,
-                    description: String::from("Bitcoin Transfer"),
-                    preimage: Some(swap_info.preimage),
-                    opening_fee_params: swap_info.channel_opening_fees,
-                    use_description_hash: Some(false),
-                    expiry: Some(SWAP_PAYMENT_FEE_EXPIRY_SECONDS),
-                    cltv: None,
-                })
-                .await?
-                .ln_invoice;
-            self.persister
-                .update_swap_bolt11(bitcoin_address.clone(), invoice.bolt11)?;
-            swap_info = self
-                .persister
-                .get_swap_info_by_address(bitcoin_address)?
-                .unwrap();
-        }
+        let bolt11 = match swap_info.bolt11 {
+            Some(bolt11) => Ok(bolt11),
+            None => {
+                // we are creating an invoice for this swap if we didn't do it already
+
+                // Try to create invoice
+                let create_invoice_res = self
+                    .payment_receiver
+                    .receive_payment(ReceivePaymentRequest {
+                        amount_msat: swap_info.confirmed_sats * 1_000,
+                        description: String::from("Bitcoin Transfer"),
+                        preimage: Some(swap_info.preimage),
+                        opening_fee_params: swap_info.channel_opening_fees,
+                        use_description_hash: Some(false),
+                        expiry: Some(SWAP_PAYMENT_FEE_EXPIRY_SECONDS),
+                        cltv: None,
+                    })
+                    .await;
+
+                // Get the invoice, depending on the creation result
+                match create_invoice_res {
+                    // Extract created invoice
+                    Ok(create_invoice_response) => {
+                        let bolt11 = create_invoice_response.ln_invoice.bolt11;
+                        self.persister
+                            .update_swap_bolt11(bitcoin_address, bolt11.clone())?;
+                        Ok(bolt11)
+                    }
 
-        // Making sure the invoice amount matches the on-chain amount
-        let payreq = swap_info.bolt11.unwrap();
-        let ln_invoice = parse_invoice(payreq.clone())?;
-        debug!("swap info confirmed = {}", swap_info.confirmed_sats);
-        if ln_invoice.amount_msat.unwrap() != (swap_info.confirmed_sats * 1000) {
-            warn!(
-                "invoice amount doesn't match confirmed sats {:?}",
-                ln_invoice.amount_msat.unwrap()
-            );
-            return Err(anyhow!("Does not match confirmed sats"));
-        }
+                    // If the swap was crated on a different device and the invoice payment failed,
+                    // the invoice already exists. In this case, lookup the invoice from the node.
```

Comment:
For me this makes sense, but I'm not sure if you mean: "...invoice payment failed..." on the other device
```suggestion
                    // If the swap was created on a different device and the invoice creation failed,
                    // then the invoice already exists. In this case, lookup the invoice from the node.
```

Reply:
Good point, it's ambiguous. I rewrote it to clarify: ec09de9b6da3e61ff09badb4b679db875557a3d9